lightning/src/chain/channelmonitor.rs

```diff
@@ -1209,23 +1209,15 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 	///
 	/// (C-not exported) because we have no HashMap bindings
 	pub fn get_outputs_to_watch(&self) -> &HashMap<Txid, Vec<Script>> {
-		&self.outputs_to_watch
-	}
-
-	/// Gets the sets of all outpoints which this ChannelMonitor expects to hear about spends of.
-	/// Generally useful when deserializing as during normal operation the return values of
-	/// block_connected are sufficient to ensure all relevant outpoints are being monitored (note
-	/// that the get_funding_txo outpoint and transaction must also be monitored for!).
-	///
-	/// (C-not exported) as there is no practical way to track lifetimes of returned values.
-	pub fn get_monitored_outpoints(&self) -> Vec<(Txid, u32, &Script)> {
-		let mut res = Vec::with_capacity(self.counterparty_commitment_txn_on_chain.len() * 2);
-		for (ref txid, &(_, ref outputs)) in self.counterparty_commitment_txn_on_chain.iter() {
-			for (idx, output) in outputs.iter().enumerate() {
-				res.push(((*txid).clone(), idx as u32, output));
+		#[cfg(any(test, feature = "fuzztarget"))]
+		for (txid, &(_, ref outputs)) in self.counterparty_commitment_txn_on_chain.iter() {
+			let watched_outputs = self.outputs_to_watch.get(txid).expect("Counterparty commitment txn which have been broadcast should have outputs registered");
+			assert_eq!(watched_outputs.len(), outputs.len());
+			for (watched, output) in watched_outputs.iter().zip(outputs.iter()) {
+				assert_eq!(watched, output);
 			}
```

Comment:
Are the order of these lists guaranteed to match? Guess this is dropped in the next commit, so may not matter.

Reply:
They should be, at least modulo #653. I did separate it out so that at least its visible that the current tests don't have things in different order.